app/controllers/users_controller.rb

```diff
@@ -14,26 +14,18 @@ def new
 
   def create
     @user = User.new(user_params)
-
-    if @user.save
-
-      begin
-        Stripe::Charge.create(card: params[:stripeToken], amount: 999, description: "Myflix charge for #{@user.email}", currency: 'usd') if Stripe.api_key
-      rescue Stripe::CardError => e
-        flash.now[:danger] = e.message
+    if @user.valid?
+      response = StripeWrapper::Charge.create(card: params[:stripeToken], description: "Myflix charge for #{@user.email}")
+      if response.successful?
+        @user.save
+        AppMailer.delay.send_mail_on_register(@user.id)
+        flash.now[:success] = 'You have been charged successfully. An email has been sent to your email. Enjoy Myflix!'
+        accept_invitation(params[:token]) if params[:token]
```

Comment:
You don't need to pass `params[:token]` to this method. `params` will be available in whole Controller class.